src/main/java/io/cdap/plugin/snowflake/common/client/SnowflakeAccessor.java

```diff
@@ -88,7 +91,11 @@ public List<SnowflakeFieldDescriptor> describeQuery(String query) throws IOExcep
         fieldDescriptors.add(new SnowflakeFieldDescriptor(name, type, nullable));
       }
     } catch (SQLException e) {
-      throw new IOException(e);
+      String errorMessage = String.format("Error occurred while executing query to fetch descriptors. " +
+        "Failed reason is %s.", e.getMessage());
```

Comment:
`Error occurred while executing query to fetch descriptors with message: %s`

This comment applies to whole PR.